include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -6662,7 +6662,57 @@ namespace VectorTools
                               norm, weight, exponent);
   }
 
+  template <int dim, int spacedim, class InVector>
+  double error_norm_from_cell_errors(const Triangulation<dim,spacedim> &tria,
+                                     const InVector &cellwise_error,
+                                     const NormType &norm)
+  {
+    Assert( cellwise_error.size() == tria.n_active_cells(),
+            ExcMessage("input vector cell_error has invalid size!"));
+
+    MPI_Comm comm = MPI_COMM_SELF;
+#ifdef DEAL_II_WITH_MPI
+    if (const parallel::Triangulation<dim,spacedim> *ptria =
+          dynamic_cast<const parallel::Triangulation<dim,spacedim>*>(&tria))
+      comm = ptria->get_communicator();
+#endif
+
+    switch (norm)
+      {
+      case L2_norm:
+      case H1_seminorm:
+      case H1_norm:
+      case Hdiv_seminorm:
+      {
+        double local = cellwise_error.l2_norm();
+        return std::sqrt (Utilities::MPI::sum (local * local, comm));
+      }
+
+      case L1_norm:
+      {
+        double local = cellwise_error.l1_norm();
+        return Utilities::MPI::sum (local, comm);
+      }
+
+      case Linfty_norm:
+      {
+        double local = cellwise_error.linfty_norm();
+        return Utilities::MPI::max (local, comm);
+      }
 
+      /* TODO:
```

Comment:
if we agree on the comment above, one can remove `TODO` and the commented block.